Grow.c

```diff
@@ -3068,6 +3069,39 @@ static void catch_term(int sig)
 	sigterm = 1;
 }
 
+/**
+ * handle_forking() - Handle forking, helper function for reshapes.
+ *
+ * @forked: is process forked.
+ * @devname: device name.
+ * @service: wheather a service should be started or just fork.
+ *
+ * Returns: -1 if failed to fork,
+ *           0 if currently child process,
+ *           1 process forked, passed to service or delegated to udev.
+ */
+static int handle_forking(bool forked, char *devname, bool service)
+{
+	/* Nothing to do if already forked. */
+	if (forked)
+		return 0;
+
+	/* Just fork if no service. */
+	if (service) {
+		/* Allow udev to trigger service after switchroot. */
```

Comment:
This is allowing anything, this is blocking service from beeing started in initramfs. that is all.

As a result of that, reshape can be always successfully started after switch root.

Reply:
it requires additional explanation why only systemd way is affected.
Maybe instead bool `use_systemd` we should check if systemd is configured on this system and adjust the behavior for both ways.

If there is no systemd, then I would like to have similar logic for forking path.

Reply:
If you are concerned about `who` will start switch root later in the case without systemd- it is not your problem. It is problem of system manager.

They can check details and call grow-continue if needed.